internal/k8shandler/cluster.go

```diff
@@ -318,24 +318,30 @@ func (er *ElasticsearchRequest) populateNodes() error {
 	minMasterUpdated := false
 
 	// we want to only keep nodes that were generated and purge/delete any other ones...
-	for _, node := range nodes[nodeMapKey(cluster.Name, cluster.Namespace)] {
-		if _, ok := containsNodeTypeInterface(node, currentNodes); !ok {
-			if !minMasterUpdated {
-				// if we're removing a node make sure we set a lower min masters to keep cluster functional
-				if er.AnyNodeReady() {
-					er.updateMinMasters()
-					minMasterUpdated = true
+	// make sure cluster is green/yellow before we delete nodes
+	if status, _ := er.esClient.GetClusterHealthStatus(); utils.Contains(desiredClusterStates, status) {
```

Comment:
nit: invert the logic to reduce nesting

Reply:
we can't invert to help out here... but also rereading it this isn't functioning as desired.... i'll update this and keep your comment in mind when fixing it